src/main/java/org/openlmis/notification/web/usercontactdetails/UserContactDetailsController.java

```diff
@@ -249,6 +303,21 @@ public LocalizedMessage verifyContactDetail(@PathVariable("id") UUID userId,
         .localize(new Message(EMAIL_VERIFICATION_SUCCESS, verificationToken.getEmailAddress()));
   }
 
+  /**
+   * Deletes user contact details.
+   *
+   * @param userIds user ids for which contact details will be deleted
+   * @return empty content
+   */
+  @DeleteMapping(value = "/userContactDetails/batch")
+  @ResponseStatus(HttpStatus.OK)
+  public ResponseEntity<Void> deleteContactDetailsByUserIds(@RequestBody Set<UUID> userIds) {
+    emailVerificationTokenRepository.deleteByUserIds(userIds);
```

Comment:
Now I see why repository needs `@Transactional` - I think the proper way would be to make a method in a Service, that method annotated with `@Transactional` calls the Repository, and is called here - some additional code definitely, but should be inline with best practices.